inc/class-group.php

```diff
@@ -45,6 +45,21 @@ public function get_title() {
 		return $this->title;
 	}
 
+	/**
+	 * Set the group title. Only update if not empty.
+	 *
+	 * @param string $title The new title.
+	 *
+	 * @return string The updated or original title.
+	 */
+	public function set_title( string $title ) : string {
+		if ( ! empty( $title ) ) {
+			$this->title = $title;
+		}
+
+		return $this->title;
```

Comment:
Do setters normally return the value too? Not sure if this is needed. Also I don't think we need to worry about the value being empty here either as you're checking if the meta value is empty before calling this method. I think it could just be a simple setter. It's unlikely to happen but if you wanted to intentionally call `$group->set_title( '' )` it would't do what you might expect.

Reply:
I tend to automatically return current values from setters so that you can check whether your call worked (eg trying to set an illegal value). It's an automatic pattern of mine, but we don't need to use it here.
However, given it is used as the anchor text for a link, it should never be empty.

Reply:
You could check the call worked with a subsequent call to the getter if really needed. Any check to see if it was an illegal value can be mitigated by type hinting, and would otherwise need to be done externally anyway if looking at the return value, unless it threw an exception. Seems like the setter has 2 jobs this way.

We're not checking the return value from it at the moment so I'd prefer to leave out anything superfluous, less code to read!

Regarding it never being empty it could be used in more places than even just anchors, and any kind of fallback output for an empty value might be best handled in each specific context. I still don't think we strictly need the check here as we're checking the value before we pass it to `set_title()` later on.

If we wanted to ensure there's always something returned I'd be more tempted to handle that in the getter e.g.

```
function get_title( ?string $fallback = null ) : string {
    return $this->title ?: $fallback ?: __( '(no title)' );
}
```

Just getting my thoughts across here so not going to require the change but it's good to discuss approaches and why we do things the way we do.

Reply:
Happy to take it out in this case because there are no rules that say the title can't be empty. But a setter validating the input so no illegal values can be set is a common pattern.